plugin/src/main/java/com/redhat/jenkins/plugins/ci/messaging/ActiveMqMessagingWorker.java

```diff
@@ -432,12 +432,17 @@ public SendResult sendMessage(Run<?, ?> build, TaskListener listener, ProviderDa
                 message.setText(PluginUtils.getSubstitutedValue(pd.getMessageContent(), envVars2));
 
 
-                publisher.send(message);
-                log.info("Sent " + pd.getMessageType().toString() + " message for job '" + build.getParent().getName() + "' to topic '" + ltopic + "':\n"
+                String messageType = "unknown type";
+                if (pd.getMessageType() != null) {
+                    messageType = pd.getMessageType().toString();
+                }
+                log.info("Send " + messageType + " message for job '" + build.getParent().getName() + "' to topic '" + ltopic + "':\n"
                         + formatMessage(message));
+                publisher.send(message);
 
                 mesgId = message.getJMSMessageID();
                 mesgContent = message.getText();
+                log.info("MessageID: " + mesgId);
```

Comment:
Should we combine this message with the one above?  Otherwise will we ever be able to link the message ID with the previous message?  I guess that means the top message would move to after the send.

Reply:
I had wanted to output to build log. my next commit will improve it.

Reply:
actually messageId is generated *after* message is sent.